scripts/zkai_attention_kv_proof_route_selector_gate.py

```
@@ -3035,6 +3256,87 @@ def validate_d16_quantized_softmax_receipt(summary: Any) -> None:
 
 
+def validate_d16_two_head_quantized_softmax_receipt(summary: Any) -> None:
+    """Validate the d16 two-head implementation-exact quantized Softmax-table receipt summary."""
+
+    if not isinstance(summary, dict):
+        raise AttentionKvRouteSelectorError("d16 two-head quantized Softmax receipt must be an object")
+    expected = d16_two_head_quantized_softmax_receipt_summary(
+        load_d16_two_head_quantized_softmax_receipt_payload()
+    )
+    if summary != expected:
+        raise AttentionKvRouteSelectorError("d16 two-head quantized Softmax receipt drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)